inference-engine/ie_bridges/python/sample/speech_sample/speech_sample.py

```diff
@@ -0,0 +1,390 @@
+                      help='Optional. Read reference score file and compare scores.')
+    args.add_argument('-d', '--device', default='CPU', type=str,
+                      help='Optional. Specify a target device to infer on. '
+                      'CPU, GPU, MYRIAD, GNA_AUTO, GNA_HW, GNA_SW_FP32, GNA_SW_EXACT and HETERO with combination of GNA'
+                      ' as the primary device and CPU as a secondary (e.g. HETERO:GNA,CPU) are supported. '
+                      'The sample will look for a suitable plugin for device specified. Default value is CPU.')
+    args.add_argument('-bs', '--batch_size', default=1, type=int, help='Optional. Batch size 1-8 (default 1).')
+    args.add_argument('-qb', '--quantization_bits', default=16, type=int,
+                      help='Optional. Weight bits for quantization: 8 or 16 (default 16).')
+    args.add_argument('-wg', '--export_gna_model', type=str,
+                      help='Optional. Write GNA model to file using path/filename provided.')
+    # TODO: Find a model that applicable for -we option
+    args.add_argument('-we', '--export_embedded_gna_model', type=str, help=argparse.SUPPRESS)
+    # TODO: Find a model that applicable for -we_gen option
+    args.add_argument('-we_gen', '--embedded_gna_configuration', default='GNA1', type=str, help=argparse.SUPPRESS)
+    args.add_argument('-iname', '--input_layers', type=str,
+                      help='Optional. Layer names for input blobs. The names are separated with ",". '
+                      'Allows to change the order of input layers for -i flag. Example: Input1,Input2')
+    args.add_argument('-oname', '--output_layers', type=str,
+                      help='Optional. Layer names for output blobs. The names are separated with ",". '
+                      'Allows to change the order of output layers for -o flag. Example: Output1:port,Output2:port.')
+
+    return parser.parse_args()
+
+
+def get_scale_factor(matrix: np.ndarray) -> float:
+    """Get scale factor for quantization using utterance matrix"""
+    # Max to find scale factor
+    target_max = 16384
+    max_val = np.max(matrix)
+    if max_val == 0:
+        return 1.0
+    else:
+        return target_max / max_val
+
+
+def read_ark_file(file_name: str) -> dict:
+    """Read utterance matrices from a .ark file"""
+    def read_key(input_file: IO[Any]) -> str:
+        """Read a identifier of utterance matrix"""
+        key = ''
+        while True:
+            char = input_file.read(1).decode()
+            if char in ('', ' '):
+                break
+            else:
+                key += char
+
+        return key
+
+    def read_matrix(input_file: IO[Any]) -> np.ndarray:
+        """Read a utterance matrix"""
+        header = input_file.read(5).decode()
+        if 'FM' in header:
+            num_of_bytes = 4
+            dtype = 'float32'
+        elif 'DM' in header:
+            num_of_bytes = 8
+            dtype = 'float64'
```

Comment:
what `else`?

Reply:
So more clearly, in my opinion.

Reply:
@dpigasin I'm not against `elif`, I'm just asking why `else` is not processed when no `FM` or `DM` in header?

Reply:
@nkogteva I'll fix it.